src/storage/storage_util.cpp

```diff
@@ -0,0 +1,100 @@
+#include <unordered_map>
+#include "storage/storage_util.h"
+#include "storage/tuple_access_strategy.h"
+#include "storage/delta_record.h"
+
+namespace terrier::storage {
+void StorageUtil::WriteBytes(uint8_t attr_size, uint64_t val, byte *pos) {
+  switch (attr_size) {
+    case sizeof(uint8_t):*reinterpret_cast<uint8_t *>(pos) = static_cast<uint8_t>(val);
+      break;
+    case sizeof(uint16_t):*reinterpret_cast<uint16_t *>(pos) = static_cast<uint16_t>(val);
+      break;
+    case sizeof(uint32_t):*reinterpret_cast<uint32_t *>(pos) = static_cast<uint32_t>(val);
+      break;
+    case sizeof(uint64_t):*reinterpret_cast<uint64_t *>(pos) = static_cast<uint64_t>(val);
+      break;
+    default:
+      // Invalid attr size
+      throw std::runtime_error("Invalid byte write value");
+  }
+}
+
+uint64_t StorageUtil::ReadBytes(uint8_t attr_size, const byte *pos) {
+  switch (attr_size) {
+    case sizeof(uint8_t):return *reinterpret_cast<const uint8_t *>(pos);
+    case sizeof(uint16_t):return *reinterpret_cast<const uint16_t *>(pos);
+    case sizeof(uint32_t):return *reinterpret_cast<const uint32_t *>(pos);
+    case sizeof(uint64_t):return *reinterpret_cast<const uint64_t *>(pos);
+    default:
+      // Invalid attr size
+      throw std::runtime_error("Invalid byte write value");
+  }
+}
+
+void StorageUtil::CopyWithNullCheck(const byte *from, ProjectedRow *to, uint8_t size, uint16_t col_id) {
```

Comment:
`void StorageUtil::CopyWithNullCheck(const byte *const from, ProjectedRow *const to, const uint8_t size, const uint16_t col_id)`